torch/ao/quantization/experimental/quantizer.py

```diff
@@ -57,15 +57,15 @@ def quantize(self, tensor2quantize: Tensor):
         result: fp representation of input Tensor
     """
     def dequantize(self, apot_tensor) -> Tensor:
-        apot_tensor_data = apot_tensor.data
+        apot_tensor_data = apot_tensor.data.flatten()
 
         # map apot_to_float over tensor2quantize elements
-        result_temp = np.empty(apot_tensor_data.size())
-        for ele in apot_tensor_data:
-            new_ele = apot_to_float(ele, self.quantization_levels, self.level_indices)
-            np.append(result_temp, new_ele)
+        result_temp = np.empty(shape=apot_tensor_data.size())
+        for i in range(len(apot_tensor_data)):
+            new_ele = apot_to_float(apot_tensor_data[i], self.quantization_levels, self.level_indices)
+            result_temp[i] = new_ele
 
-        result = torch.from_numpy(result_temp).int()
+        result = torch.from_numpy(result_temp)
 
         return result
```

Comment:
seems like the output of this for a 3x3 tensor would be a 1x9 tensor, shouldn't this output a tensor with the same output dimensions as the input tensor?

can you add a unit test for this?

Reply:
yes agreed, fixed!